File: completable-reactor-runtime/src/main/java/ru/fix/completable/reactor/runtime/gl/GlCoordinatesImpl.java
package ru.fix.completable.reactor.runtime.gl;

final class GlCoordinatesImpl implements GlCoordinates {
    @Override
    public GlCoordinates payload(int x, int y) {
        return this;
    }

    @Override
    public GlCoordinates handler(Vertex vertex, int x, int y) {
        return this;
    }

    @Override
    public GlCoordinates merger(Vertex vertex, int x, int y) {
        return this;
    }

    @Override
    public GlCoordinates router(Vertex vertex, int x, int y) {
        return this;
    }

    @Override
    public GlCoordinates complete(Vertex vertex, int x, int y) {
        return this;
    }
}

File: completable-reactor-runtime/src/main/kotlin/ru/fix/completable/reactor/runtime/gl/GlCoordinatesImpl.kt
package ru.fix.completable.reactor.runtime.gl

internal class GlCoordinatesImpl : GlCoordinates {
    override fun payload(x: Int, y: Int): GlCoordinates {
        return this
    }

    override fun handler(vertex: Vertex, x: Int, y: Int): GlCoordinates {
        return this
    }

    override fun merger(vertex: Vertex, x: Int, y: Int): GlCoordinates {
        return this
    }

    override fun router(vertex: Vertex, x: Int, y: Int): GlCoordinates {
        return this
    }

    override fun complete(vertex: Vertex, x: Int, y: Int): GlCoordinates {
        return this
    }

    override fun subgraph(vertex: Vertex, x: Int, y: Int): GlCoordinates {
        return this
    }
}
